pipeline/flows/notifications.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+from typing import Dict, List
+
+import prefect
+from prefect import Flow, Parameter
+from prefect.tasks.secrets import PrefectSecret
+
+from pipeline.extract import extract_users
+from pipeline.notifications import NewMessageNotifier
+from pipeline.transform import convert_users_from_df
+from pipeline.types import ChatData, NotificationInfo, NotificationInput
+from pipeline.utils.firebase import initialize_firebase_for_prefect
+
+
+@prefect.task
+def get_users_to_notify(inp: NotificationInput) -> List[NotificationInfo]:
+    logger = prefect.context.get("logger")
+    # add other types of notifiers once they are implemented
+    notifier = NewMessageNotifier()
+    logger.info("Generating notifications")
+    # refactor to account for additional notification types once they're implemented
+    notif_info_lst = notifier.get_notifications(inp=inp)
+    logger.info(
+        f"Done. Number of notifications generated: {len(notif_info_lst)}"
+    )
+    return notif_info_lst
```

Comment:
Looks good! You could start by creating a list of notifiers, even if right now, it will only have one notifier in it. Then you can adapt your logic to loop through the notifiers and generate all the notifications.